### PR TITLE
fix: allowed `LatLngBounds.center` to work across world boundary & added `simpleCenter`

### PR DESCRIPTION
**Pull request by monsieurtanuki**
### What
* The formula referenced for the `LatLngBounds.center` mentioned a [-180, +180] fix - now that fix is implemented.
* A more intuitive `LatLngBounds.simpleCenter` is now implemented.

### Impacted files
* `latlng_bounds.dart`: fixed the computation of `center` according to referenced link; added a more intuitive `simpleCenter` method.
* `latlng_bounds_test.dart`: added the tests that failed in the initial issue

### CODE REVIEW

File: lib/src/geo/latlng_bounds.dart
Comment (monsieurtanuki) [reply lib/src/geo/latlng_bounds.dart:180]: I found that fix in the referenced link.
The idea is to get a clean -180+180 longitude in order to avoid the longitude assert crash.

**Comment by JaffaKetchup** [comment]
@josxha Did you have any other concerns about this?

**Review by JaffaKetchup — APPROVED** [review]
LGTM, although maybe a little bit of better documentation would be good? Maybe a diagram of where the two centers would calculate different values, and which is correct to use for which situation?
Will wait for @josxha to approve before merge.

**Comment by josxha** [comment]
The changes in this pull request are straight forward. The only two things I struggle with are the name `simpleCenter`. I'd prefer a more descriptive way but haven't found one ever since. And the other thing is that we should use this new center calculation for the camera, shouldn't we?

**Review by josxha — APPROVED** [review]
Alright 👍

**Pull request merged**
